tests.py

```diff
@@ -116,318 +139,494 @@ def test_version_number_in_setup(self):
+            self.api.authenticate(self.user_name, self.terminal_name)
         self.assertRaises(toopher.ToopherApiError, fn)
 
     def test_unrecognized_error_still_raises_error(self):
-        api = toopher.ToopherApi('key', 'secret')
-        api.client = HttpClientMock({
+        self.api.advanced.raw.client = HttpClientMock({
             'authentication_requests/initiate': (409,
-                json.dumps({'error_code': 42,
-                            'error_message': 'what'}))})
+                json.dumps({
+                    'error_code': 42,
+                    'error_message': 'what'
+                })
+            )
+        })
 
         def fn():
-            api.authenticate_by_user_name('user_name', 'terminal_name')
+            self.api.authenticate(self.user_name, self.terminal_name)
         self.assertRaises(toopher.ToopherApiError, fn)
 
-class ZeroStorageTests(unittest.TestCase):
-    def test_create_user_terminal(self):
-        api = toopher.ToopherApi('key', 'secret')
-        api.client = HttpClientMock({'user_terminals/create': (200, '{}')})
-
-        api.create_user_terminal('user_name', 'terminal_name', 'requester_terminal_id')
-
-        last_called_data = api.client.last_called_data
-        self.assertEqual(api.client.last_called_method, 'POST')
-        self.assertEqual(last_called_data['user_name'], 'user_name')
-        self.assertEqual(last_called_data['name'], 'terminal_name')
-        self.assertEqual(last_called_data['name_extra'], 'requester_terminal_id')
-
-    def test_enable_toopher_for_user(self):
-        api = toopher.ToopherApi('key', 'secret')
-        api.client = HttpClientMock({
-            'users': (200, json.dumps([{'id': 'user_id', 'name': 'user_name'}])),
-            'users/user_id': (200, json.dumps({'name': 'user_name'}))})
-
-        api.set_toopher_enabled_for_user('user_name', True)
-        self.assertEqual(api.client.last_called_method, 'POST')
-        self.assertFalse(api.client.last_called_data['disable_toopher_auth'])
-
-        api.set_toopher_enabled_for_user('user_name', False)
-        self.assertEqual(api.client.last_called_method, 'POST')
-        self.assertTrue(api.client.last_called_data['disable_toopher_auth'])
-
-    def test_enable_toopher_multiple_users(self):
-        api = toopher.ToopherApi('key', 'secret')
-        api.client = HttpClientMock({'users': (200,
-            json.dumps([{'name': 'first user'}, {'name': 'second user'}]))})
-
-        def fn():
-            api.set_toopher_enabled_for_user('multiple users', True)
-        self.assertRaises(toopher.ToopherApiError, fn)
-
-    def test_enable_toopher_no_users(self):
-        api = toopher.ToopherApi('key', 'secret')
-        api.client = HttpClientMock({'users': (200, '[]')})
-
-        def fn():
-            api.set_toopher_enabled_for_user('no users', True)
-        self.assertRaises(toopher.ToopherApiError, fn)
 
+class ZeroStorageTests(unittest.TestCase):
```

Comment:
I would vote for merging these tests with the other `pair` and `authenticate` tests. Since we don't have a ZeroStorage class in our code (I believe it's only mentioned in the README), I don't believe it's necessary to make this distinction in the tests.